### PR TITLE
Navi escaped commas and quotes

### LINKED ISSUE #1046: Support escaped values in Elide filters

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #1046 

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/facts/elide.ts [hunk 1]
```diff
@@ -23,6 +23,14 @@ import { v1 } from 'ember-uuid';
 
 export const ELIDE_API_DATE_FORMAT = 'YYYY-MM-DD'; //TODO: Update to include time when elide supports using full iso date strings
 
+const escape = (value: string) => {
+  const fn = (v: string) => `${('' + v).replace(/'/g, "\\'").replace(/,/g, '\\,')}`;
+  if (Array.isArray(value)) {
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:28]: Do we have this case?

File: packages/data/tests/unit/adapters/facts/elide-test.ts [hunk 2]
```diff
@@ -524,4 +524,53 @@ module('Unit | Adapter | facts/elide', function(hooks) {
...
+  test('escaped filter values', async function(assert) {
+    const adapter: ElideFactsAdapter = this.owner.lookup('adapter:facts/elide');
+    const EscapedTest: RequestV2 = {
+      table: 'table1',
+      columns: [
+        { field: 'table1.m1', type: 'metric', parameters: {} },
+        { field: 'table1.m2', type: 'metric', parameters: {} },
+        { field: 'table1.r', type: 'metric', parameters: { p: '123', as: 'a' } },
+        { field: 'table1.d1', type: 'dimension', parameters: {} },
+        { field: 'table1.d2', type: 'dimension', parameters: {} }
```
Comment (jkusa) [inline packages/data/tests/unit/adapters/facts/elide-test.ts:537]:
```suggestion
```
Do we need all these example columns?

File: packages/data/tests/unit/adapters/facts/elide-test.ts [hunk 3]
```diff
@@ -524,4 +524,53 @@ module('Unit | Adapter | facts/elide', function(hooks) {
...
+          type: 'dimension',
+          operator: 'in',
+          values: ['with "quote"', 'but why']
+        },
+        {
+          field: 'table1.d8',
+          parameters: { field: 'id' },
+          type: 'dimension',
+          operator: 'in',
+          values: ["with 'single quote'", 'okay']
```
Comment (jkusa) [inline packages/data/tests/unit/adapters/facts/elide-test.ts:559]:
```suggestion
          values: ['okay', "with 'single quote'"]
```
Just to make sure the escaping works on values other than the first index :)

File: packages/data/addon/adapters/facts/elide.ts (hunk 1 above)
Comment (holderb5) [reply packages/data/addon/adapters/facts/elide.ts:28]: For this situation I do not think so, I'll remove it

File: packages/data/tests/unit/adapters/facts/elide-test.ts (hunk 2 above)
Comment (holderb5) [reply packages/data/tests/unit/adapters/facts/elide-test.ts:537]: No, I was just going off the other test request, I'll delete them

File: packages/data/addon/adapters/facts/elide.ts [hunk 4]
```diff
@@ -23,6 +23,11 @@ import { v1 } from 'ember-uuid';
 
 export const ELIDE_API_DATE_FORMAT = 'YYYY-MM-DD'; //TODO: Update to include time when elide supports using full iso date strings
 
+const escape = (value: string) => {
+  const fn = (v: string) => `${('' + v).replace(/'/g, "\\'").replace(/,/g, '\\,')}`;
+  return fn(value);
+};
```
Comment (kevinhinterlong) [inline packages/data/addon/adapters/facts/elide.ts:29]:
```suggestion
const escape = (value: string) => {
  return `${('' + value).replace(/'/g, "\\'").replace(/,/g, '\\,')}`;
};
```

**Review by kevinhinterlong — APPROVED** [review]

**Pull request merged**